AtD: Use `admin_url()` correctly

mdawaffe opened the pull request:
Use `admin_url( "admin-ajax.php…" )` instead of `admin_url() . "admin-ajax.php…"`. When filters are attached to `admin_url`, the concatenation pattern may no longer work.

Syncs file with WP.com: D23473-code.

To test that things still work:
1. Turn on AtD (Jetpack -> Settings -> Writing -> Composing -> "Check your spelling, style, and grammar").
2. Under "Ignored Phrases", add a nonsense word: `yaay`.
3. Click the Composing block's "Save Settings" button.
4. Create a new post in the Classic Editor.
5. Type the nonsense word in the content. (You'll probably see your browser highlight it as a misspelling.)
6. Click "proofread" in the Text editor or "ABC✔️" in the Visual editor.
7. See that the nonsense word is *not* highlighted as a misspelling by AtD.

Commit:
AtD: Use `admin_url()` correctly

Summary:
Use `admin_url( "admin-ajax.php…" )` instead of `admin_url() . "admin-ajax.php…"`. When filters are attached to `admin_url`, the concatenation pattern may no longer work.

Syncs file with WP.com: D23473-code.

To test that things still work:
1. Turn on AtD (Jetpack -> Settings -> Writing -> Composing -> "Check your spelling, style, and grammar").
2. Under "Ignored Phrases", add a nonsense word: `yaay`.
3. Click the Composing block's "Save Settings" button.
4. Create a new post in the Classic Editor.
5. Type the nonsense word in the content. (You'll probably see your browser highlight it as a misspelling.)
6. Click "proofread" in the Text editor or "ABC✔️" in the Visual editor.
7. See that the nonsense word is *not* highlighted as a misspelling by AtD.

jeherve reviewed (dismissed):
This seems to work in my tests. However, I get a different error in the console when trying to fix mistakes using the AtD tools:
https://videopress.com/v/7MvNMQK5

`Uncaught TypeError: suggestionsMenu.hideMenu is not a function`
in `modules/after-the-deadline/tinymce/plugin.js?v=20150715&wp-mce-4800-20180716`

The error seems different from what was reported in #9128

mdawaffe committed:
AtD: Switch to "new" menu's API.

`tinymce.ui.Menu()`'s `.hidemenu()` is from TinyMCE v3.

The new (v4) API seems to be `.hide()`.

mdawaffe commented:
I think the error is fixed as of 254090eee. Will require a separate WP.com sync.

jeherve reviewed (approved):
This works well. Merging.
